library/src/main/java/io/ona/kujaku/helpers/storage/TrackingStorage.java

```diff
@@ -0,0 +1,91 @@
+package io.ona.kujaku.helpers.storage;
+
+import android.location.Location;
+import android.os.Environment;
+import android.util.Log;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Emmanuel OTIN - eo@novel-t.ch on 12/03/2019.
+ *
+ */
+public class TrackingStorage extends BaseStorage {
```

Comment:
Can we rather save it in an accessible DB probably with the API to fetch? Rather than a single file for each point. What do you think?

Reply:
I use a single file for each point for a performance matter. The gson library doesn't allow you to "append" a point in a file representing a list of objects. So you need to rewrite the full collection each time you get a point. I did tests on a long distance during more than 2 hours driving and saw that the app is lagging when you need to write a big collection every 2 seconds.

Reply:
😲  Aaah! The rationale behind using a DB is that these are records and not specifically blobs. Therefore we need to easily query them like any other records. The DB is a data management system because it handles all the issues that are related to such data management without someone having to re-address them.

You are true in that saving to a single file is just a bit faster since they are all being saved to storage anyway. 

The assumption is that the host application will require these records and upload them to a server. Right? And for specific periods or rather specific outdoor missions(or as they are updated locally). I already have moving around around 21 files in that folder like 20x20 metres space.

Does that make more sense? Does that win you over?

Reply:
I understand but all permanent storage is the host app responsability with the different business logics. The library only store de last "route" and provide the collection of Locations to the app.

Reply:
OK, Sorry, so you actually provide the recorded locations. Nice, but why still save them on local storage?

Reply:
We are saving the current collection of locations and the previous collection. Just to be able to retrieve them after having stopped the Tracking Service. The host application can manage crashes or other exceptions and then retrieve last saved locations.

Reply:
Cool